ariadne/cardano/test/Test/Ariadne/Bip44.hs

```diff
@@ -0,0 +1,147 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Test.Ariadne.Bip44
+       ( bip44PathGen
+       , bip44KeyPairGen
+       ) where
+
+import Test.Hspec(Spec, describe)
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (Property, property, (===), conjoin, (.&&.))
+import Test.QuickCheck.Arbitrary (Arbitrary(..))
+import Test.QuickCheck.Gen (Gen(..))
+
+import Ariadne.Wallet.Cardano.Kernel.Bip44
+
+import Ariadne.Wallet.Cardano.Kernel.Word31 (Word31, unsafeMkWord31, word31ToWord32)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet 
+  ( HdAddressChain (..)
+  , HdAccountIx (..)
+  , HdAddressIx(..))
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Derivation (deriveBip44KeyPairTest)
+
+import Pos.Crypto.HD (firstNonHardened, firstHardened)
+
+import Pos.Core (Address(..), IsBootstrapEraAddr(..), makePubKeyHdwAddress)
+import Cardano.Crypto.Wallet (generate, toXPub, unXPrv, xprv, xpub)
+import Data.Default
+import qualified Pos.Crypto as S 
+import Data.ByteArray
+import qualified Data.ByteString.Base16 as B16 
+
+instance Arbitrary Bip44DerivationPath where
+    arbitrary = do
+      bip44AccountIndex <- arbitrary
+      bip44AddressChain <- arbitrary
+      bip44AddressIndex <- arbitrary
+      return Bip44DerivationPath {..}
+
+{-------------------------------------------------------------------------------
+  Constants
+-------------------------------------------------------------------------------}
+
+-- | BIP-44 derivation constant
+bip44Purpose :: Word31
+bip44Purpose = unsafeMkWord31 44

```

Comment:
Perhaps it is better to export this value from `Ariadne.Wallet.Cardano.Kernel.Bip44` rather than copy it here.

Reply:
Ok.